internal/core/src/mmap/Column.h

```diff
@@ -286,10 +287,108 @@ class VariableColumn : public ColumnBase {
         views_.emplace_back(data_ + indices_.back(), size_ - indices_.back());
     }
 
- private:
     std::vector<uint64_t> indices_{};
 
     // Compatible with current Span type
     std::vector<ViewType> views_{};
 };
+
+class ArrayColumn : public ColumnBase {
+ public:
+    // memory mode ctor
+    ArrayColumn(size_t num_rows,
+                const FieldMeta& field_meta,
+                const DataType element_type)
+        : ColumnBase(num_rows, field_meta), element_type_(element_type) {
+    }
+
+    // mmap mode ctor
+    ArrayColumn(const File& file,
+                size_t size,
+                const FieldMeta& field_meta,
+                const DataType element_type)
+        : ColumnBase(file, size, field_meta), element_type_(element_type) {
+    }
+
+    ArrayColumn(ArrayColumn&& column) noexcept
+        : ColumnBase(std::move(column)),
+          indices_(std::move(column.indices_)),
+          views_(std::move(column.views_)),
+          element_type_(column.element_type_) {
+    }
+
+    ~ArrayColumn() override = default;
+
+    SpanBase
+    Span() const override {
+        return SpanBase(views_.data(), views_.size(), sizeof(Array));
+    }
+
+    [[nodiscard]] const std::vector<Array>&
+    Views() const {
+        return views_;
+    }
+
+    Array
+    operator[](const int i) const {
+        return views_[i];
+    }
+
+    ScalarArray
+    RawAt(const int i) const {
+        auto array = views_[i];
+        return array.output_data();
+    }
+
+    void
+    Append(const char* data, size_t size) {
+        indices_.emplace_back(size_);
+        ColumnBase::Append(data, size);
+    }
+
+    void
+    Append(const char* data,
+           size_t size,
+           DataType element_type,
+           const std::vector<uint64_t>& element_indices) {
+        indices_.emplace_back(size_);
+        element_type_ = element_type;
+        element_indices_.emplace_back(element_indices);
+        ColumnBase::Append(data, size);
+    }
+
+    void
+    Seal(std::vector<uint64_t> indices = {},
+         std::vector<std::vector<uint64_t>> element_indices = {}) {
+        if (!indices.empty()) {
+            indices_ = std::move(indices);
+            element_indices_ = std::move(element_indices);
+        }
+        ConstructViews();
+    }
+
+ protected:
+    void
+    ConstructViews() {
+        views_.reserve(indices_.size());
+        for (size_t i = 0; i < indices_.size() - 1; i++) {
+            views_.emplace_back(data_ + indices_[i],
+                                indices_[i + 1] - indices_[i],
+                                element_type_,
+                                element_indices_[i]);
+        }
+        views_.emplace_back(data_ + indices_.back(),
+                            size_ - indices_.back(),
+                            element_type_,
+                            element_indices_[indices_.size() - 1]);
```

Comment:
for (size_t i = 0; i < indices_.size(); i++) {
            views_.emplace_back(data_ + indices_[i],
                               i == indices_.size() -1 ? size_ - indices_[i] : indices_[i + 1] - indices_[i],
                                element_type_,
                                element_indices_[i]);
        }
can simply code, but need  check performance